crux-lucene/test/crux/lucene_test.clj

```diff
@@ -0,0 +1,200 @@
+      (with-open [before-db (c/open-db *api*)]
+        (submit+await-tx [[:crux.tx/put {:crux.db/id :ivan2 :name "Ivbn"}]])
+        (let [q {:find '[?e] :where '[[(text-search "Iv?n" :name) [[?e]]] [?e :crux.db/id]]}]
+          (t/is (= #{[:ivan]} (c/q before-db q)))
+          (with-open [db (c/open-db *api*)]
+            (t/is (= #{[:ivan] [:ivan2]} (c/q db q)))))))
+
+    (t/testing "Modifying doc"
+      (with-open [before-db (c/open-db *api*)]
+        (submit+await-tx [[:crux.tx/put {:crux.db/id :ivan :name "Derek"}]])
+        (let [q {:find '[?e] :where '[[(text-search "Derek" :name) [[?e]]] [?e :crux.db/id]]}]
+          (t/is (not (seq (c/q before-db q))))
+          (with-open [db (c/open-db *api*)]
+            (t/is (= #{[:ivan]} (c/q db q)))))))
+
+    (t/testing "Eviction"
+      (submit+await-tx [[:crux.tx/put {:crux.db/id :ivan2 :name "Derek"}]])
+      (submit+await-tx [[:crux.tx/evict :ivan]])
+      (with-open [db (c/open-db *api*)]
+        (t/is (empty? (c/q db {:find '[?e]
+                               :where
+                               '[[(text-search "Ivan" :name) [[?e]]]
+                                 [?e :crux.db/id]]}))))
+      (with-open [search-results ^crux.api.ICursor (l/search (:crux.lucene/node @(:!system *api*)) "name" "Ivan")]
+        (t/is (empty? (iterator-seq search-results))))
+      (with-open [search-results ^crux.api.ICursor (l/search (:crux.lucene/node @(:!system *api*)) "name" "Derek")]
+        (t/is (seq (iterator-seq search-results)))))
+
+    (t/testing "Scores"
+      (submit+await-tx [[:crux.tx/put {:crux.db/id "test0" :name "ivon"}]])
+      (submit+await-tx [[:crux.tx/put {:crux.db/id "test1" :name "ivan"}]])
+      (submit+await-tx [[:crux.tx/put {:crux.db/id "test2" :name "testivantest"}]])
+      (submit+await-tx [[:crux.tx/put {:crux.db/id "test3" :name "testing"}]])
+      (submit+await-tx [[:crux.tx/put {:crux.db/id "test4" :name "ivanpost"}]])
+      (with-open [db (c/open-db *api*)]
+        (t/is (= #{["test1" "ivan" 1.0] ["test4" "ivanpost" 1.0]}
+                 (c/q db {:find '[?e ?v ?score]
+                          :where '[[(text-search "ivan*" :name) [[?e ?v _ ?score]]]
+                                   [?e :crux.db/id]]})))))
+
+    (t/testing "cardinality many"
+      (submit+await-tx [[:crux.tx/put {:crux.db/id :ivan :foo #{"atar" "abar" "nomatch"}}]])
+
+      (with-open [db (c/open-db *api*)]
+        (t/is (= #{[:ivan "atar"]}
+                 (c/q db {:find '[?e ?v]
+                          :where '[[(text-search "atar" :foo) [[?e ?v]]]
+                                   [?e :crux.db/id]]}))))
+
+      (with-open [db (c/open-db *api*)]
+        (t/is (= #{[:ivan "abar"]
+                   [:ivan "atar"]}
+                 (c/q db {:find '[?e ?v]
+                          :where '[[(text-search "a?ar" :foo) [[?e ?v]]]
+                                   [?e :crux.db/id]]})))))))
+
+(t/deftest test-can-search-string-across-attributes
```

Comment:
good that we _could_ do this, but worth mandating the attribute to keep it consistent with the rest of the query engine?
maybe allowing multiple search attributes if that's what this is required for?

Reply:
Moved out into a different pred-fn